presto-hive/src/main/java/io/prestosql/plugin/hive/ViewReader.java

```diff
@@ -0,0 +1,252 @@
+                HiveToRelConverter hiveToRelConverter = HiveToRelConverter.create(mscClient);
+                RelNode rel = hiveToRelConverter.convertView(table.getDatabaseName(), table.getTableName());
+                RelToPrestoConverter rel2Presto = new RelToPrestoConverter();
+                String prestoSql = rel2Presto.convert(rel);
+                RelDataType rowType = rel.getRowType();
+                List<ConnectorViewDefinition.ViewColumn> columns = new ArrayList<>(rowType.getFieldCount());
+                for (RelDataTypeField field : rowType.getFieldList()) {
+                    log.debug("Adding column %s, %s, %s, %s", field.getName(),
+                            field.getType().getFullTypeString(),
+                            field.getType().getSqlTypeName(),
+                            field);
+
+                    Type type = typeManager.fromSqlType(getTypeString(field.getType()));
+                    ConnectorViewDefinition.ViewColumn column = new ConnectorViewDefinition.ViewColumn(field.getName(), type.getTypeId());
+                    columns.add(column);
+                }
+                return new ConnectorViewDefinition(prestoSql,
+                        Optional.of(catalogName.toString()),
+                        Optional.of(table.getDatabaseName()),
+                        columns,
+                        Optional.ofNullable(table.getParameters().get(TABLE_COMMENT)),
+                        Optional.empty(),
+                        true);
+            }
+            catch (RuntimeException e) {
+                throw new PrestoException(HIVE_UNKNOWN_ERROR,
+                        format("Error decoding view definition for %s.%s: %s. Please report this to ask_dali@linkedin.com",
+                                table.getDatabaseName(),
+                                table.getTableName(),
+                                e.getMessage()),
+                        e);
+            }
+        }
+
+        // Calcite does not provide correct type strings for non-primitive types.
+        // We add custom code here to make it work. Goal is for calcite/coral to handle this
+        private String getTypeString(RelDataType type)
+        {
+            switch (type.getSqlTypeName()) {
+                case ROW: {
+                    Preconditions.checkState(type.isStruct());
+                    StringBuilder sb = new StringBuilder("row(");
+                    List<RelDataTypeField> fieldList = type.getFieldList();
+                    for (int i = 0; i < fieldList.size(); i++) {
+                        if (i != 0) {
+                            // There should not be space after comma. Presto deserializer doesn't like it
+                            sb.append(",");
+                        }
+                        RelDataTypeField field = fieldList.get(i);
+                        sb.append(field.getName().toLowerCase(Locale.ENGLISH)).append(" ").append(getTypeString(field.getType()));
```

Comment:
Does this work for field names with special characters that need quoting? Or does the `field` already return the quoted name?

Reply:
Yes. It works with field names with special characters.
I tested with following example with field name "__a_" and field name "count": 
> select * from hive.u_ychen6.viewtest_view3;
 __a_ | _b_ 
------+-----
    1 |   1 
    2 |   2 
> select * from hive.u_ychen6.viewtest_view2;
 col1 | count 
------+-------
    1 |     1 
    2 |     2

Reply:
`_` doesn't need quoting AFAIK. Maybe some column with `.` in it's name?

Reply:
Sorry, I was wrong. RelDataTypeField does not return quoted names.

Reply:
@martint Any way to get properly quoted identifiers (or to quote them) for such cases?

The end-goal seems to be to generate DDL like `ROW(plain_column, "dotted.column.name")`.

Reply:
@hashhar, you'd need to have rules that understand what characters are not valid in the context of an identifier and quote them. Alternatively, you can quote all generated identifiers and avoid having to think about it.

Reply:
Quoted identifiers have different semantics when you consider case sensitivity.

Reply:
Indeed, but presumably, something that knows how to translate queries automatically would generate identifiers that are already normalized according to the semantics of the target.